### PR TITLE
Revert #1635 + fix for #1633

### PR DESCRIPTION
**Pull request by rgrinberg**

### CODE REVIEW

**Comment by jonludlam** [comment]
Thanks for this!

File: src/process.ml [hunk 1]
```diff
@@ -30,9 +30,67 @@ let map_result
...
+    ; fd = lazy fd
+    ; channel = lazy stdout
+    ; status = Keep_open
+    }
+  let stdout = terminal stdout
+  let stderr = terminal stderr
+
+  let file fn =
+    let fd =
+      lazy (Unix.openfile (Path.to_string fn)
```
Comment (jonludlam) [inline src/process.ml]: Presumably this has the same issue I was worried about - that by the time this is actually executed we might have changed dir. I wasn't at all convinced it was necessary, we certainly call `Output.fd` before `Scheduler.with_chdir`, but are there any other paths by which cwd might change?

**Review by jonludlam — APPROVED** [review]
Other than that question it looks good to me.

File: src/process.ml (hunk 1 above)
Comment (diml) [reply src/process.ml]: Nope, we do all the chdirs at a single place in Dune. I added a comment to make sure we don't move the Output.fd calls inside the `with_chdir`.

**Pull request merged**